model-server/src/main/kotlin/org/modelix/model/server/handlers/RepositoriesManager.kt

```diff
@@ -145,12 +146,19 @@ class RepositoriesManager(val client: LocalModelClient) : IRepositoriesManager {
      * For backwards compatibility, existing repositories remain in the global storage.
      */
     override fun isIsolated(repository: RepositoryId): Boolean? {
-        // The repository might not exist, but new repositories will be created with isolated storage.
-        // If a repository is not part of the legacy ones it's considered isolated.
-        if (getRepositories(true).contains(repository)) return true
-        if (getRepositories(false).contains(repository)) return false
-        return null
+        isIsolatedCache[repository]?.let { return it }
+        val result = when {
+            getRepositories(true).contains(repository) -> true
+            getRepositories(false).contains(repository) -> false
+            else -> null
+        }
+        if (result != null) {
+            // cache invalidation not necessary because the value isn't expected to change after it's known
```

Comment:
Someone could remove an isolated repository and recreate it as a non-isolated repository?
This could happen if someone creates an isolated repository, but then notices that he wants to use the V1 API and needs and a non-isolated repository.

Reply:
Reverted that change. There is no caching `isIsolated` anymore.